### PR TITLE
Make incompatible command-line args an error

### PR DESCRIPTION
**Pull request by ravenblackx**
Commit Message: Make incompatible command-line args an error
Additional Description: Per #35073, `--enable-fine-grain-logging` and `--component-log-level` don't make sense at the same time, one of them will be overridden by the other in a way that can be surprising for the user. This PR makes it an error to apply both, so the user is not surprised by not getting the logs they expected at runtime.
Risk Level: Small. May make existing command-lines with ineffective options fail.
Testing: Added coverage.
Docs Changes: Updated command-line options docs.
Release Notes: Yes, under minor behavior changes.
Platform Specific Features: n/a

### CODE REVIEW

File: source/server/options_impl.cc [hunk 1]
```diff
@@ -214,7 +214,12 @@ OptionsImpl::OptionsImpl(std::vector<std::string> args,
   log_format_ = log_format.getValue();
   log_format_set_ = log_format.isSet();
   log_format_escaped_ = log_format_escaped.getValue();
+
   enable_fine_grain_logging_ = enable_fine_grain_logging.getValue();
+  if (enable_fine_grain_logging_ && !component_log_level.getValue().empty()) {
+    throw MalformedArgvException(
+        std::string{"error: --component-log-level will not work with --enable-fine-grain-logging"});
```
Comment (botengyao) [inline source/server/options_impl.cc:221]: nit: std::string can be removed?

**Review by botengyao — COMMENTED** [review]
Thanks for the improvement!

Could you also add a change-log for this change?

/wait

**Review by botengyao — APPROVED** [review]
Nice, thanks!

Do we need a second pass from other folks?

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**